models/mongo/notes.go

```diff
@@ -0,0 +1,89 @@
+package mongo
+
+import (
+	"accounts-service/models"
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.uber.org/zap"
+	"golang.org/x/net/context"
+)
+
+type notesRepository struct {
+	logger *zap.Logger
+	db     *mongo.Database
+	coll   *mongo.Collection
+}
+
+func NewNotesRepository(db *mongo.Database, logger *zap.Logger) models.NotesRepository {
+	rep := &notesRepository{
+		logger: logger.Named("mongo").Named("notes"),
+		db:     db,
+		coll:   db.Collection("notes"),
+	}
+
+	_, err := rep.coll.Indexes().CreateOne(
+		context.Background(),
+		mongo.IndexModel{
+			Keys:    bson.D{{Key: "note_id", Value: 1}, {Key: "group_id", Value: 1}},
```

Comment:
What do you think about this change? This makes sure that the same note isn't assigned to two different groups. If a student wants to share one note between multiple groups he will need to make a copy.
```suggestion
			Keys:    bson.D{{Key: "note_id", Value: 1}},
```